### ISSUE TITLE
[Bug] Unable to build with different compilers and platforms after bumping `ntex` version

### ISSUE THREAD

**Issue by evshary** [eclipse-zenoh/zenoh-plugin-mqtt#223]
### Describe the bug

The package failed in the [release process](https://github.com/eclipse-zenoh/zenoh-plugin-mqtt/actions/runs/11023955108). You can see there are some failures with different compilers and platforms.
![image](https://github.com/user-attachments/assets/84dcf14a-6823-4ca3-a6e9-48a7702ab7b6)
These failures come from `rustls` inside the `ntex`. Since `rustls` is using `aws-lc-rs` by default and `ntex` doesn't provide an option to disable it and use `ring` instead, we can only try to dive into why `aws-lc-rs` failed.

After some analysis, it's caused by different issues:

1. Unable to build with `x86_64-unknown-linux-gnu` on Ubuntu 20.04: GCC is too old in 20.04 and needs to update.
2. Unable to build with ARM compiler on Ubuntu  20.04: Need to add the `bindgen` feature in `aws-lc-rs`
3. Unable to build with `x86_64-pc-windows-gnu` on Windows-2019: GCC is too old in 2019 and must install NASM to avoid using the prebuilt one.

In short, we need these steps inside the CI to upgrade GCC.
https://github.com/evshary/zenoh-plugin-mqtt/blob/b96def9f9836a70ebaef246d6ddaecce7ce702cf/.github/workflows/ci.yml#L38-L55

After some discussion, we decided to put off the upgrade of `ntex` until the Zenoh 1.0 release, which means will need to revert the `ntex` upgrade commit for the time being.
In the meantime, we will also see if it's possible to use `ring` inside `ntex`. (https://github.com/ntex-rs/ntex/issues/428)

### To reproduce

None.

### System info

None.

**Comment by evshary** [comment]
Close after the PR is merged and pass the dry-run release.
https://github.com/eclipse-zenoh/zenoh-plugin-mqtt/pull/228
